extensions/amp-subscriptions/0.1/platform-store.js

```diff
@@ -366,68 +366,114 @@ export class PlatformStore {
   }
 
   /**
-   * Returns most qualified platform. Qualification of a platform is based on an
-   * integer weight. Every platform starts with weight 0 and evaluated against
+   * Calculates weight to add/remove based on getSupportedScoreFactor()
+   * @param {string} factorName
+   * @param {!./subscription-platform.SubscriptionPlatform} platform
+   * @return {!Promise<number>}
+   * @private
+   */
+  getSupportedFactorWeight_(factorName, platform) {
+    return platform.getSupportedScoreFactor(factorName)
+        .then(factorValue => {
+          if (typeof factorValue !== 'number') {
+            return 0;
+          }
+          return this.scoreConfig_[factorName] *
```

Comment:
Is it guaranteed for `this.scoreConfig_[factorName]` to always exist? Should it be defaulted to 0?

Reply:
Yes it will always exists because this is called by iterating this.scoreConfig_ at L464.  The check for `number` if defensive coding for the case were some async even in a service fails and we get something weird like an `Error` or `unknown` back.